### ISSUE TITLE
systemd-networkd: Failed to set CAN interface configurations: No such device

### ISSUE THREAD

**Issue by marckleinebudde** [systemd/systemd#20651]
**systemd version the issue has been seen with**
 > v249

<!-- **NOTE:** Do not submit bug reports about anything but the two most recently released *major* systemd versions upstream! -->
<!--           If there have been multiple stable releases for that major version, please consider updating to a recent one before reporting an issue. -->
<!--           When using a distro package, please make sure that the version reported is meaningful for upstream. -->
<!-- See https://github.com/systemd/systemd-stable/releases for the list of most recent releases. -->
<!-- For older version please use distribution trackers (see https://systemd.io/CONTRIBUTING#filing-issues). -->

**Used distribution**
 > debian

**Linux kernel version used** (`uname -a`)
<!-- Make sure to enclose the pasted kernel version in `backticks`, so that
     GitHub doesn't convert the `#` character typically included in it into a
     reference to old GitHub issues. -->
 > `Linux rpi4 5.14.0-rc5-v8+ #53 SMP PREEMPT Tue Aug 17 19:22:51 CEST 2021 aarch64 GNU/Linux`

**CPU architecture issue was seen on**
 > aarch64

**Expected behaviour you didn't see**
 > CAN interfaces are configured properly

**Unexpected behaviour you saw**
 > CAN interfaces are _not_ configured

**Steps to reproduce the problem**
 > use `systemd-networkd` to configure CAN interfaces:
 > `/etc/systemd/network/70-mcp251xfd0.link`
```
[Match]
Driver=mcp251xfd
Path=platform-3f204000.spi-cs-0 platform-fe204000.spi-cs-0

[Link]
Name=mcp251xfd0
```

 > `/etc/systemd/network/80-can.network`:
```
[Match]
Name=*can* mcp251x*

[CAN]
BitRate=500K
RestartSec=1s
```
Start `systemd-networkd`.

**Additional program output to the terminal or log subsystem illustrating the issue**

<!-- Please paste relevant program terminal or journal output here, ideally
     when generated in debug mode (try setting the SYSTEMD_LOG_LEVEL=debug
     environment variable). For very long copy/pasted data consider using a
     service like https://gist.github.com/. Where copy/paste is not possible
     (for example early boot or late shutdown), a photo of the screen might do
     too, but text is always much preferred. -->
```text
lo: Link state is up-to-date                                                                                                                                                                                                                                    
lo: State changed: initialized -> unmanaged                                                                                                                                                                                                                     
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_31 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=14 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a                     
mcp251xfd0: Failed to set CAN interface configurations: No such device                                                                                                                                                                                          
mcp251xfd0: Failed                                                                                                                                                                                                                                              
mcp251xfd0: State changed: configuring -> failed                                                                                                                                                                                                                
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_37 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=15 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a                     
mcp251xfd1: Failed to set CAN interface configurations: No such device                                                                                                                                                                                          
mcp251xfd1: Failed                                                                                                                                                                                                                                              
mcp251xfd1: State changed: configuring -> failed                                                                                                                                                                                                                
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_36 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=16 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a                     
wlan0: IPv6LL address generation mode set.                                                                                                                                                                                                                      
wlan0: master interface set.                                                                                                                                                                                                                                    
wlan0: link_check_ready(): link is not activated.                                                                                                                                                                                                               
wlan0: Bringing link up                                                                                                                                                                                                                                         
```

I've bisected the problem to commit:
```
commit 7558f9e717381eef0ddc8ddfb5a754ea4b0f3e6c (HEAD, refs/bisect/bad)
Author: Yu Watanabe <watanabe.yu+github@gmail.com>
Date:   Sun Jun 6 14:14:44 2021 +0900

    network: use request queue to configure CAN interfaces
    
    This also makes SR-IOV configurations are ignored for CAN interfaces,
    as CAN interfaces seem not to support SR-IOV features.
```

**Comment by yuwata** [comment]
Already fixed by 1e8cce8f1e61e01db844d518b7051b6ce69867fd, and the commit is already backported to v249-stable repository 9bf50758426a3d8dd4b40e28c960e920d41444ba, which is included in v249.4.

Duplicate of #20428. Closing.

**Comment by marckleinebudde** [comment]
Just found that out myself and bisecting for the fix - sorry for the noise.
